pkg/types/ccipocr3/plugin_execute_types.go

```diff
@@ -11,3 +17,35 @@ type ExecutePluginReportSingleChain struct {
 	Proofs              []Bytes32     `json:"proofs"`
 	ProofFlagBits       BigInt        `json:"proofFlagBits"`
 }
+
+type CommitReportWithMessages struct {
+	MerkleRootChain
+	MessagesToExecute []Message
+}
+
+// ExecuteReportInfo contains metadata needed by transmitter and contract
+// writer.
+type ExecuteReportInfo []MerkleRootChain
+
+// Encode v1 execute report info. Very basic versioning in the first byte to
+// allow for future encoding optimizations.
+func (eri ExecuteReportInfo) Encode() ([]byte, error) {
+	data, err := json.Marshal(eri)
+	data = append([]byte{1}, data...)
```

Comment:
```suggestion
	data = append([]byte{01}, data...)
```
2 digits just in case